packages/web/src/utils/interpolate-env-variables-to-index.ts

```diff
@@ -0,0 +1,52 @@
+export function interpolateEnvironmentVariablesToIndex(
+  contents: string,
+  publicUrl?: string
+): string {
+  const environmentVariables = getClientEnvironment(publicUrl || '');
+  return interpolateEnvironmentVariables(contents, environmentVariables as any);
+}
+
+const NX_PREFIX = /^NX_/i;
+
+function isNxEnvironmentKey(x: string): boolean {
+  return NX_PREFIX.test(x);
+}
+
+function getClientEnvironment(publicUrl: string) {
+  return Object.keys(process.env)
+    .filter(isNxEnvironmentKey)
+    .reduce(
+      (env, key) => {
+        env[key] = process.env[key];
+        return env;
+      },
+      {
+        NODE_ENV: process.env.NODE_ENV || 'development',
+        PUBLIC_URL: publicUrl || process.env.PUBLIC_URL || '',
+        FAST_REFRESH: process.env.FAST_REFRESH !== 'false',
```

Comment:
`FAST_REFRESH` isn't used by us so we can remove here. For CRA they use it to pass to the dev server. When we integrate fast refresh we can add it back if needed. :)